### ISSUE TITLE
Site Editor: Cover Block Uses Image But Still Requests Premium Upgrade for Video

### ISSUE THREAD

**Issue by jeyip** [Automattic/wp-calypso#50407]
## Description
When uploading a cover block, the "Upgrade to Premium" message is rendered, asking us to upgrade because we're using a video. This happens even if we're _not_ uploading a video.

![2021-02-23 14 34 53](https://user-images.githubusercontent.com/5414230/108917836-3c9fd380-75e5-11eb-81d4-aff930fb7a82.gif)

## Steps to reproduce the behavior

1. Navigate to the site editor
2. Add a cover block
3. Upload an image
4. Verify that the “Upgrade your plan to use video covers” appears

#### What I expected to happen
The upgrade message isn't displayed until I upload a video cover

#### What actually happened
The message is displayed regardless of what I upload

**Comment by simison** [comment]
@apeatling could this be on the View backlog? It relates only to the site editor and not to the post editor. Checking how compatible rest of the aid blocks and their nudges would be good, too.

**Comment by andrewserong** [comment]
I can only recreate this in production, and not from my sandbox, where additional assets aren't loaded from s0,s1,s2, etc. If I update my sandbox to use those instead, I can recreate the issue. So I'm wondering if this is to do with loading additional JS / styles from within the editor canvas iframe (I haven't verified this yet, it's just a hunch at the end of my day 🙂)? If so, I'm not sure which change will need to be made, or where to do it, but happy to keep digging. Just a heads-up @jeyip — it might take me a little while to get to the bottom of this, since I'm not that familiar with how the site editor works / where the JS & CSS is being loaded, but I'll continue investigating.

More context for the sandbox difference in p7H4VZ-2qv-p2

**Comment by simison** [comment]
Could sandboxing/not-sandboxing the store have an effect on this?

The code for the upsell should be in Jetpack methinks.

You can create a [site for FSE](https://horizon.wordpress.com/new?flags=gutenboarding/site-editor) and access site-editor (vs page/post editor) that way.

**Comment by andrewserong** [comment]
It turns out the upsell itself is a bit of a red herring — I think it might actually be a CSS loading issue. The `jetpack-blocks-editor-css` style isn't getting copied over to the site editor's iframe because it doesn't exist in `window.__editorStyles.html` and it also doesn't get grabbed from the parent document because when it's loaded externally to the site the stylesheet doesn't have a `.cssRules` property ([here's where it's tested](https://github.com/WordPress/gutenberg/blob/748421d1980b1aeb486283d90d09a9bfefde068d/packages/block-editor/src/components/iframe/index.js#L32) for the property in Gutenberg).

To get the Jetpack editor styles picked in [client-assets.php](https://github.com/WordPress/gutenberg/blob/9484b1ed1dd60ad9cbc721e91e35f4b7b5ed7143/lib/client-assets.php#L698), we need to make sure that the editor styles are being registered along with at least one of the Jetpack blocks that gets registered. I've managed to get this working in my sandbox, just gotta tidy it up and get it working properly, but I think I've got a solution in https://github.com/Automattic/jetpack/pull/19197 🤞 — I'll follow up over there tomorrow.

**Comment by andrewserong** [comment]
Update: https://github.com/Automattic/jetpack/pull/19197 has been merged, but not yet deployed to wpcom. I'll keep this issue open until it lands.

**Comment by andrewserong** [comment]
The fix has now been deployed to wpcom, so the upgrade nudge isn't displayed for the Cover block by default, thanks to the CSS loading fix that also fixes styling for other Jetpack blocks (e.g. Donations, Contact Form).

I noticed a separate issue, though, which is that the upgrade nudge behaviour when you click to upload a video doesn't appear to be working. In the post or page editor, if you select a video file to upload, it cancels the upload and then displays the nudge:

![image](https://user-images.githubusercontent.com/14988353/112262775-4be86e80-8cc2-11eb-86e9-cb6a2eaf82c6.png)

However in the site editor, it proceeds to attempt to upload and gives the not-at-all-friendly security error message instead:

![image](https://user-images.githubusercontent.com/14988353/112262874-73d7d200-8cc2-11eb-9537-14b24f2ed342.png)

I'll leave this GH issue open and follow up on this next issue, and see how hard it'll be to fix up for the site editor.

**Comment by andrewserong** [comment]
@jeyip I managed to come up with a fix for the Cover block video uploading issue where it displays the server-based error message instead of overriding the error message in Jetpack before the file is uploaded. It turns out that the Site Editor in Gutenberg isn't server rendering the `allowedMimeTypes` setting, whereas this does get rendered for the post and page editor. I've submitted a fix over in Gutenberg in https://github.com/WordPress/gutenberg/pull/30436 — to generalize it for Gutenberg I used an `svg` file as an example, however I've tested the fix in a wpcom sandbox, and afterwards applying it, it correctly displays the following when attempting to upload a video file while on the free plan:

![image](https://user-images.githubusercontent.com/14988353/113253196-a2802900-9310-11eb-9836-93b9d71d242b.png)

**Comment by jeyip** [comment]
Thanks for following up on this @andrewserong! It looks like I'm late to the party because https://github.com/WordPress/gutenberg/pull/30436 was merged, but I really appreciate the help. If anything else comes up, I'll do my best to respond a bit faster next time 👍

**Comment by andrewserong** [comment]
Thanks @jeyip, not a worry at all! This one wasn’t urgent and got reviewed and merged in nice and quickly 😊

**Comment by davipontesblog** [comment]
Hi @andrewserong , doing some triaging for issues with "[Status] Fix Inbound" here 👋  Is this one still waiting on any merges to Dotcom?

**Comment by andrewserong** [comment]
Thanks for checking in @davipontesblog. This one is waiting on Gutenberg v10.4.0 to land in dotcom (I see it's included in [10.4.0 RC1](https://github.com/WordPress/gutenberg/releases/tag/v10.4.0-rc.1)), so once that version lands, we can close this issue out.

**Comment by andrewserong** [comment]
Closing now as this change has been rolled out to wpcom (as of GB 10.4.1)
